crates/sui-core/src/epoch/reconfiguration.rs

```diff
@@ -97,20 +117,38 @@ impl<A> ActiveAuthority<A> {
         );
         self.state
             .change_epoch_tx
-            .store(Some(Arc::new(advance_epoch_tx)));
+            .store(Some(Arc::new(advance_epoch_tx.clone())));
+
+        // Collect a certificate for this system transaction that changes epoch,
+        // and execute it locally.
+        loop {
+            if let Ok(certificate) = new_net
+                .process_transaction(
+                    advance_epoch_tx.clone().to_transaction(),
+                    Duration::from_secs(0),
+                )
+                .await
+            {
+                self.state
+                    .handle_confirmation_transaction(ConfirmationTransaction { certificate })
+                    .await
+                    .expect("Executing the special cert cannot fail");
+                break;
+            }
 
-        // TODO: Now ask every validator in the committee for this signed tx.
-        // Aggregate them to obtain a cert, execute the cert, and then start the new epoch.
+            tokio::time::sleep(WAIT_BETWEEN_EPOCH_TX_QUERY_RETRY).await;
```

Comment:
Good catch. Changed the mechanism a bit: now a validator will simply store the signed tx to the db as if it's from the normal flow. And at anytime anyone asking for this tx, it will be able to respond with the signed tx (and potentially also the cert)